core/orchestrator.go

```diff
@@ -653,10 +656,13 @@ func (n *LivepeerNode) transcodeSegmentLoop(logCtx context.Context, md *SegTrans
 
 func (n *LivepeerNode) endTranscodingSession(sessionId string, logCtx context.Context) {
 	// timeout; clean up goroutine here
-	if n.StorageConfig != nil {
-		n.StorageConfig.OS.EndSession()
-		n.StorageConfig.LocalOS.EndSession()
+	n.storageMutex.Lock()
+	if storage, exists := n.StorageConfigs[sessionId]; exists {
+		storage.OS.EndSession()
```

Comment:
This seems like an OS operation which may take a moment, then would it maybe be possible to execute outside of the lock? Otherwise, you may experience other goroutines waiting on this lock.

Reply:
StorageConfigs need thread safety because called by go routine [here](https://github.com/livepeer/go-livepeer/blob/80e70628ed1cace7a88784a37dc5312bb86663bf/core/orchestrator.go#L638) and grpc protocol.